### ISSUE TITLE
`readonly-type` Is triggered by empty object

### ISSUE THREAD

**Issue by RebeccaStevens** [eslint-functional/eslint-plugin-functional#796]
```ts
{} // Readonly type using 'readonly' keyword is forbidden. Use 'Readonly<T>' instead
```
